realm-transformer/src/main/groovy/io/realm/transformer/BytecodeModifier.groovy

```diff
@@ -85,6 +85,13 @@ class BytecodeModifier {
         clazz.addInterface(proxyInterface)
     }
 
+    public static void callInjectObjectContextFromDefaultConstructor(CtClass clazz) {
+        def defaultConstructor = clazz.getDeclaredConstructor()
+        defaultConstructor.insertBeforeBody('if ($0 instanceof io.realm.internal.RealmObjectProxy) {' +
```

Comment:
This doesn't match the code you posted in the description 😉 . Instead of the instance check would it make more sense to have the method being part of getter/setter interface we add to all model classes anyway? That way you could call `realm$injectObjectContext` directly since it would just be a no-op for unmanaged classes? 

EDIT: http://stackoverflow.com/questions/103564/the-performance-impact-of-using-instanceof-in-java I guess either approach is fine so probably no reason to change this.